data/common/__init__.py

```diff
@@ -212,3 +212,38 @@ def fix_unit(unit):
             return "t⋅km"
         case _:
             return unit
+
+
+def format_number(num):
+    """Format a number to a string using python general format and simplified exponential notation.
+
+    Args:
+        num: A number (int or float) to format
+
+    Returns:
+        str: Formatted number string with:
+        - 6 significant digits (using Python's general format)
+        - Simplified exponential notation to conform to prettier requirements where:
+            - "e+0" becomes "e" (e.g., "1.23e+05" → "1.23e5")
+            - "e-0" becomes "e-" (e.g., "1.23e-05" → "1.23e-5")
+            - "e+" becomes "e" (e.g., "1.23e+5" → "1.23e5")
+    """
+    if isinstance(num, (int, float)):
+        # Convert to scientific notation with 6 significant digits
+        num_py_g = f"{num:.6g}"
+        # Clean up the exponential notation
+        return float(
+            num_py_g.replace("e+0", "e").replace("e-0", "e-").replace("e+", "e")
```

Comment:
Yes that's true. The purpose of the replace was for the test function to rigorously the same tests `1e-4` and not `1e-04`. But that's no use since we can't  produce numbers in python without like `1e-4`. We have to produce `1e-04` and let prettier reformat it to `1e-4`